cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -153,18 +154,16 @@ bool DrawableSubmap::MaybeFetchTexture(ros::ServiceClient* const client) {
   query_in_progress_ = true;
   last_query_timestamp_ = now;
   rpc_request_future_ = std::async(std::launch::async, [this, client]() {
-    ::cartographer_ros_msgs::SubmapQuery srv;
-    srv.request.trajectory_id = id_.trajectory_id;
-    srv.request.submap_index = id_.submap_index;
-    if (client->call(srv)) {
+    std::unique_ptr<::cartographer_ros::SubmapTexture> submap_texture =
+        ::cartographer_ros::FetchSubmapTexture(id_, client);
+    ::cartographer::common::MutexLocker locker(&mutex_);
+    query_in_progress_ = false;
```

Comment:
Add a comment in the destructor about the logic there: `query_in_progress_` must be true until the Q_EMIT happened. Qt then makes sure that RequestSucceeded() is not executed if this object gets destroyed in the meantime.